### PR TITLE
UTF-8 Passthough

### PR DESCRIPTION
**Pull request by rustyrussell**
I enhanced my summry plugin to use UTF-8 for drawing bars, but it gets mangled into \uxxxx.  Fixing this took all morning, and you won't like the final patch, but I'm pretty sick of Python right now...

### CODE REVIEW

File: contrib/pylightning/tests/test_version.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import lightning
+
+
+def test_version():
```
Comment (cdecker) [inline contrib/pylightning/tests/test_version.py]: We can actually ensure this programmatically: `setup.py` can run arbitrary code to detect stuff, so we could just import the `lightning` module and use `__version__` in the call to `setup()`

File: contrib/pylightning/lightning/plugin.py [hunk 2]
```diff
@@ -378,9 +378,11 @@ def _dispatch_notification(self, request):
             self.log(traceback.format_exc())
 
     def _write_locked(self, obj):
-        s = json.dumps(obj, cls=LightningRpc.LightningJSONEncoder) + "\n\n"
+        # ensure_ascii turns UTF-8 into \uXXXX so we need to suppress that,
+        # then utf8 ourselves.
+        s = bytes(json.dumps(obj, cls=LightningRpc.LightningJSONEncoder, ensure_ascii=False) + "\n\n", encoding='utf-8')
```
Comment (cdecker) [inline contrib/pylightning/lightning/plugin.py]: Good catch, I always get confused with `decode` and `encode` as well.

File: contrib/pylightning/tests/test_version.py (hunk 1 above)
Comment (rustyrussell) [reply contrib/pylightning/tests/test_version.py]: OK, great!  Wasn't sure if that would work if the dependencies for lightning module aren't installed yet, but maybe that's not what setup.py is for.

**Comment by rustyrussell** [comment]
Rebased to fold the fix, will merge post-travis.

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/2438/commits/cdc4c8b61c2249678553467264316a0094a8bca4

**Pull request merged**